src/azure-cli/azure/cli/command_modules/cdn/custom.py

```
@@ -71,6 +91,304 @@ def update_endpoint(instance,
+                           custom_fragment, source_pattern, destination, preserve_unmatched_path)
+    for i in range(0, len(policy.rules)):
+        if policy.rules[i].name == rule_name:
+            policy.rules[i].actions.append(action)
+
+    params = EndpointUpdateParameters(
+        delivery_policy=policy
+    )
+
+    return client.endpoints.update(resource_group_name, profile_name, endpoint_name, params)
+
+
+def remove_rule(client, resource_group_name, profile_name, endpoint_name, rule_name):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    if policy is not None:
+        for rule in policy.rules:
+            if rule.name == rule_name:
+                policy.rules.remove(rule)
+
+    params = EndpointUpdateParameters(
+        delivery_policy=policy
+    )
+
+    return client.endpoints.update(resource_group_name, profile_name, endpoint_name, params)
+
+
+def remove_condition(client, resource_group_name, profile_name, endpoint_name, rule_name, index):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    if policy is not None:
+        for i in range(0, len(policy.rules)):
+            if policy.rules[i].name == rule_name:
+                policy.rules[i].conditions.pop(int(index))
+
+    params = EndpointUpdateParameters(
+        delivery_policy=policy
+    )
+
+    return client.endpoints.update(resource_group_name, profile_name, endpoint_name, params)
+
+
+def remove_action(client, resource_group_name, profile_name, endpoint_name, rule_name, index):
+
+    endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
+    policy = endpoint.delivery_policy
+    if policy is not None:
+        for i in range(0, len(policy.rules)):
+            if policy.rules[i].name == rule_name:
+                policy.rules[i].actions.pop(int(index))
```

could you add exception and useful error message here?

---

If there is nothing to remove, the command will return the latest policy to customer. I don't think we need to throw an exception in this case